### ISSUE TITLE
"System.Management.Automation.PSCustomObject" to type "System.Collections.ArrayList"

### ISSUE THREAD

**Issue by MaynardMiner** [MaynardMiner/SWARM#30]
```
SWARM Is Building The Database. Auto-Coin Switching: No
Checking Algo Pools
Adding Custom Pools
Checking Algo Miners
Warning: Blocking phoenix-1 mining ethash on whalesburg for NVIDIA1
Cannot convert value "@{Delay=1; Symbol=ethash; MinerName=ethdcrminer64; Prestart=System.Object[]; Type=NVIDIA1; Path=.\bin\claymore-1\ethdcrminer64; Devices=; DeviceCall=claymore; Arguments=-platform 2 -mport 1333 -mode 1 -allcoins 1 -allpools 1 -epool stratum+ssl://eu1.whalesburg.com:7777 -ewal  -eworker swarm1_2 -wd 0 -gser 2 -dbg -1 -eres 1 ; HashRates=; Quote=0.000155247831824539; PowerX=; ocpower=190; occore=0; ocmem=1100; ocfans=; ethpill=Yes; pilldelay=5; FullName=ethash; API=claymore; Port=1333; MinerPool=whalesburg; URI=https://github.com/MaynardMiner/MM.Compiled-Miners/releases/download/v4.0/Claymore-Linux.tar.gz; BUILD=Tar; Algo=ethash; Name=claymore-1}" to type "System.Collections.ArrayList". Error: "Cannot convert the "@{Delay=1; Symbol=ethash; MinerName=ethdcrminer64; Prestart=System.Object[]; Type=NVIDIA1; Path=.\bin\claymore-1\ethdcrminer64; Devices=; DeviceCall=claymore; Arguments=-platform 2 -mport 1333 -mode 1 -allcoins 1 -allpools 1 -epool stratum+ssl://eu1.whalesburg.com:7777 -ewal  -eworker swarm1_2 -wd 0 -gser 2 -dbg -1 -eres 1 ; HashRates=; Quote=0.000155247831824539; PowerX=; ocpower=190; occore=0; ocmem=1100; ocfans=; ethpill=Yes; pilldelay=5; FullName=ethash; API=claymore; Port=1333; MinerPool=whalesburg; URI=https://github.com/MaynardMiner/MM.Compiled-Miners/releases/download/v4.0/Claymore-Linux.tar.gz; BUILD=Tar; Algo=ethash; Name=claymore-1}" value of type "System.Management.Automation.PSCustomObject" to type "System.Collections.ArrayList"."
At /root/SWARM.2.0.4/swarm.ps1:1021 char:9
```

User is reporting an issue with the conversion to a [System.Collections.Arraylist].

I have only 1 known report. I believe it is an issue with his Powershell version. I have been unable to reproduce.

https://community.spiceworks.com/topic/2050505-convert-string-for-system-collections-arraylist

According this article- Another developer experienced issue, and found the solution was to convert item to Array before adding to [System.Collections.Arraylist]. I will attempt to do so, for powershell version compatibility.

**Comment by MaynardMiner** [comment]
It was confirmed user was using Powershell 6.1.3, when SWARM is coded in 6.1.0. This may be cause of issue, and may need to be addressed before upgrading powershell core base.

**Comment by MaynardMiner** [comment]
I solved it.

1.) There is a typo for getting custom pools. Must have happened a few versions ago.

Line 26 Pool Command:

```
"Custom"{$GetPools = if (Test-Path "coinpools") {Get-ChildItemContent "coinpools" | ForEach {if($_ -ne $Null){$_.Content | Add-Member @{Name = $_.Name} -PassThru}}}}
```

should be:

```
"Custom"{$GetPools = if (Test-Path "custompools") {Get-ChildItemContent "custompools" | ForEach {if($_ -ne $Null){$_.Content | Add-Member @{Name = $_.Name} -PassThru}}}}
```

2.) Loading the Array was causing the issue. Building the container solved the problem. Took awhile to figure that out.

ChildItems.ps1 line 54 changed to:

```
    $AllContent = New-Object System.Collections.ArrayList
    $ChildItems | %{$AllContent.Add($_) | Out-Null}
    $AllContent
```

To prevent further errors of Arraylists, I am changing all instances of loading to Arraylists to the above format.

**Comment by MaynardMiner** [comment]
Proposed fix applied: 2.0.6
